### ISSUE TITLE
Popup doesn't closed after login on social networks

### ISSUE THREAD

**Issue by pirmax** [dgrubelic/vue-authenticate#77]
When I'm login in with Twitter, the popup doesn't close and the response isn't give.

Could you help me please?

**Comment by Rockfordal** [comment]
I had same problem, but got it working by starting out with the example files
in config.json enter:
    clientID, clientSecret
and in index.js:
 providers: {
    facebook: {
      clientId: "<yourid>"
also console.log()
also in server around line 117 it helps to console.log(err)

**Comment by makslevental** [comment]
popups don't close if you're not already authed with the service because of this line 

https://github.com/dgrubelic/vue-authenticate/blob/master/src/oauth/popup.js#L75

this needs to poll for redirect in popup path rather than just waiting 250ms

**Comment by makslevental** [comment]
if your app pops up a system modal (e.g. asking for camera permission) this prevents the popup from closing). my solution was to set the redirect to be a blank page (which works since vue-auth parses query params anyway)

**Comment by phamcongsonit** [comment]
Dauthenticate

**Comment by phamcongsonit** [comment]
Does anyone have a solution?

**Comment by faisaltheparttimecoder** [comment]
Thanks @makslevental , you are a saver ... that worked .. bravo ... 

Your config showed me the error that I was doing wrong, in my case my callback was to localhost & I have using the web browser with 127.0.0.1 to test my web app. changed the web browser to localhost and everything seems to work ... thanks so much @makslevental

**Comment by SeedyROM** [comment]
@makslevental Currently trying to implement your work around, but I'm running into issues with authenticate throwing a 400 after the pop-up closes.

Not entirely sure what's up. I'll post again later tonight.

**Comment by SeedyROM** [comment]
So it's been 2 weeks, and I still have no idea what's going on with the popup work around.

**Comment by faisaltheparttimecoder** [comment]
Hello @SeedyROM, not sure about the 400 , could be something is wrong at your end of the code...

Here is my working example.
 
https://github.com/faisaltheparttimecoder/carelogFrontend/blob/master/src/main.js#L49

the keys.js is not part of the repo, but it looks like this ...

```
// Set the base url based on prod or dev environment
// this is the link to the application backend URL.
const baseUrl = function () {
  if (process.env.NODE_ENV === 'production') {
    return 'https://xxxxx.com/'
  } else {  // Dev environment ( localhost )
    return 'http://127.0.0.1:8000/'
  }
}

// All the secret keys
module.exports = {
  baseUrl: baseUrl(), // Backend API URL
  google: {
    client_id: 'xxxxxxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxxx.apps.googleusercontent.com',
    client_secret: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxx',
    scope: ['profile', 'email', 'openid'],
    whitelist_domain: 'xxxxx.com'
  }
}
```

Hope you find a solution ...

**Comment by SeedyROM** [comment]
The bug only happens on mobile. I should probably open my own issue.

**Comment by sun-ergo-s** [comment]
I have the same problem with Facebook. The popup window doesn't close and authentication process continues in popup with callback URL. I've got the error message in original tab: "Auth popup window closed", even the popup is still opened. On the contrary, Google authentication works fine.

**Comment by sun-ergo-s** [comment]
Problem solved for me. I had Facebook Container extension installed in Firefox. After removing it Facebook login works.

**Comment by Ashraf514** [comment]
**Vue** 
**library**: vue-social-auth / vue-authenticate

I was facing the same issue while integrating Google and Microsoft login (using diff libraries). Pop up wasn't closing, and it was showing me the same login screen (Application's) again and again. 
After looking into it deeply I found that some navigation guard (it was checking authenticated user) was interrupting the redirection to the dashboard (post login screen). When I removed (resolved the condition issue later) that it started working.

Following is the code that might help to relate:

this.$auth.authenticate(provider).then(async res => {
                    const paramObj  =   { type: provider };
                    const payload   =   Object.assign({}, paramObj, res);
                    console.log(payload);

                    const response = await request({
                        method:'POST',
                        url:`/auth/social-login`,
                        data:payload
                    });
                    const { data } = response;
                    await setStorage("auth", JSON.stringify(data));
                    await this.$store.dispatch("user", data);
                    await this.$router.push({ name: "test" });
                }).catch(err => {
                    console.log("error",err);
                });
                
                
                
   **Navigation guard causing issue**
   
   if (
        to.matched.some(record => record.meta.authRequired)
      && !hasAuthUser()
   ) {
       removeStorage('auth');
        next({
            name: 'login',
        });
  }



So, Basically, I want to say that if you face similar issue that is nothing to do with the library, there must be some bug inside our code.
